calliope/test/test_cli.py

```diff
@@ -120,3 +123,22 @@ def test_generate_scenarios(self):
             assert scenarios['scenarios']['scenario_1'] == [
                 'cold_fusion', 'run1', 'group_share_cold_fusion_cap'
             ]
+
+    def test_no_success_exit_code_when_infeasible(self):
+        runner = CliRunner()
+        result = runner.invoke(cli.run, [
+            _MINIMAL_TEST_MODEL,
+            "--scenario=supply_purchase,investment_costs",  # without these, the model cannot run
+            "--override_dict={techs.test_supply_elec.constraints.energy_cap_max: 1}"  # elec supply too low
+        ])
+        assert result.exit_code != 0
+
+    def test_success_exit_code_when_infeasible_and_demanded(self):
+        runner = CliRunner()
+        result = runner.invoke(cli.run, [
+            _MINIMAL_TEST_MODEL,
+            "--no_fail_when_infeasible",
+            "--scenario=supply_purchase,investment_costs",  # without these, the model cannot run
```

Comment:
supply_purchase means that it is running a MILP model... I would go with 'supply_simple' to have the smallest possible model running (if that allows the model to run, ofc)

Reply:
`supply_simple` doesn't exist but is not necessary anyway: `investment_costs` alone has all we need.